src/main/java/io/seqera/migtool/builder/SqlTemplate.java

```diff
@@ -0,0 +1,22 @@
+package io.seqera.migtool.builder;
+
+/**
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ */
+public abstract class SqlTemplate {
```

Comment:
By the way, this is not a [Template](https://www.digitalocean.com/community/tutorials/template-method-design-pattern-in-java) pattern. What I had initially actually is a template pattern...

In the template pattern you define a method (or several) in the superclass that is the template method, which the subclasses can call, overriding the abstract methods. In your version, there is no such template method (as opposed to my version where in the abstract class we define the three possible queries, with the subclasses implementing the different building blocks). This is just an extension of an abstract class imo.

Reply:
The implementation is so simple that I find a bit tedious debating if it's a template or strategy pattern. Still matching the wikipedia definition imo

> The helper methods may be either abstract methods, in which case subclasses are required to provide concrete implementations, or hook methods, which have empty bodies in the superclass

Reply:
> The helper methods may be either abstract methods

Yes, the helper methods may be abstract, but you are missing the _template_ method., hence the name of the pattern.

As per Wikipedia:  
<img width="1216" alt="image" src="https://github.com/user-attachments/assets/27e75cbf-4b51-4e14-addb-ca1fa2a8dbe7">

Which is exactly the pattern my original PR proposed... :) 

I have stated issues that are _bad design_ in your solution, so I am not going to repeat them again. Hence, I still think that either of my proposals are better. Also, I added tests for all the changes, which are more complete than the current ones.

> The implementation is so simple that I find a bit tedious debating

With this I agree with you, so that is why I don't understand all of the pushback for the change? 😅 


Also, there are missing bits (like in MigRecord) that do not allow your solution to be used from Java code @pditommaso ...

Reply:
Very stimulating challenge, indeed!

Reply:
> Also, there are missing bits (like in MigRecord) that do not allow your solution to be used from

Please report a separate PR for it

Reply:
> Please report a separate PR for it

Will do.

Are you also ok if I open issues to refactor the Migtool hardcoded strings and implement a more OOP style, as in my original PR?

Reply:
I'm fine with meaningful improvements e.g. const/enums for string. I found overkill a builder class in a project with 5 classes

Reply:
> I'm fine with meaningful improvements e.g. const/enums for string. I found overkill a builder class in a project with 5 classes

There was a reason for the builder though... When extracting the Driver and Dialect internally from the Url, we need to make sure the Url is set first, if we do not use the builder. 
Another reason was to have all fields of the Migtool as final, to avoid mutating them once set, but the main driver for the change was the first one.

Reply:
Fair enough, still not builder is not needed  https://github.com/seqeralabs/migtool/pull/50/files

Reply:
> Fair enough, still not builder is not needed https://github.com/seqeralabs/migtool/pull/50/files

Yes, there are different ways to make it work, but I would prefer to use a known pattern and have all fields as immutable :)